rfcs/step_wait_promise.md

```diff
@@ -0,0 +1,87 @@
+# RFC 207: `step_wait_promise` method
+
+## Summary
+
+Add a `step_wait_promise(promise, description, timeout=default_timeout)` method to 
+the wpt `Test` object. This method waits for the given `promise` to resolve within 
+the specified `timeout`. If the promise is not resolved within the timeout period, it 
+raises an assertion error: `Timed out waiting on condition`.
+
+## Details
+
+* **`step_wait_promise(promise, description, timeout=default_timeout)`**
```

Comment:
Do we want to add a `Test.step_wait_promise_done` as well?

`Test.step_wait` already optionally takes a Promise as its first argument, so is this basically just:

```
Test.prototype.step_wait_promise(promise, description, timeout) {
    return this.step_wait_func(() => promise, description, timeout)
}
```

Reply:
indeed, this should work. Thanks!

Reply:
But perhaps this can be fixed in the `step_wait_func` method instead.